client/cody-shared/src/chat/recipes/generate-pr-description.ts

```diff
@@ -40,7 +40,10 @@ export class PrDescription implements Recipe {
             prTemplateContent = readFileSync(templatePath).toString()
         }
 
-        const gitCommit = spawnSync('git', ['log', 'origin/HEAD..HEAD', logFormat], { cwd: dirPath })
+        const userName = spawnSync('git', ['config', 'user.name'], { cwd: dirPath })
+        const user = userName.stdout.toString().trim()
+
+        const gitCommit = spawnSync('git', ['log', `--author=${user}`, logFormat], { cwd: dirPath })
```

Comment:
We should probably use the email address instead of the name, the email is more likely to be unique in a larger org.

```suggestion
        const userEmail = spawnSync('git', ['config', 'user.email'], { cwd: dirPath })
        const email = userEmail.stdout.toString().trim()

        const gitCommit = spawnSync('git', ['log', `--author=<${email}>`, logFormat], { cwd: dirPath })
```

Reply:
The user here is the `username` of git which is also unique for each user. If we use an email address, we need to add some extra commands for filtering, as the filtered log based on the email considers the complete history, not just the current branch.
So, I think using userName is also fine. LMK what do you think else I can update it to email.

Reply:
@deepak2431 Not sure I follow. `user.name` is literally just the name that I configured in git:

```
α sourcegraph (main)
git config user.name
Philipp Spiess
```

what am I missing?

> as the filtered log based on the email considers the complete history, not just the current branch.

I’m pretty sure that both approaches just go through `git log` and filter based on the name string field which happens to be `${name} <${email}>`

Reply:
Both consider the complete history, I think we want:

```
git log --author="<hello@philippspiess.com>" origin/HEAD..HEAD
```

Reply:
I used to generally config my git user id in` user.name.` Yeah, that's not the general convention.  I updated the command by including `origin/HEAD..HEAD`, which works well. I saw in some blogs some other ways to filter it out, so I asked you once about it :)

All good to go for this now!